### PR TITLE
feat(tasks): added link to job in failure mail

### PR DESCRIPTION
**Pull request by harmbrugge**
Fixes [#4177](https://github.com/molgenis/molgenis-emx2/issues/4177)
To provided more context on failing jobs a link to the job is provided in the failure mail

What are the main changes you did:
- Added the server url to ScriptTasks
- Added a link to to job in the failure email template

The title of the mail is:
`Molgenis script <script name> failed on <server name>`

The content is:
```
Molgenis script <script name> failed with error:
<error>

<link to the job>
```
Preview link:
https://preview-emx2-pr-4179.dev.molgenis.org/_SYSTEM_/tasks/#/jobs?id=9a5c51c9-d5e3-4d53-9bb2-995ae35bbc12

how to test:
- Submit a job with an error and a failureAddress
- Receive an email with a link to the job

### CODE REVIEW

**Review by mswertz — COMMENTED** [review]
do you see a way to get sonar coverage happy?

**Comment by mswertz** [comment]
I noticed when not signed in I get endless spinner when I follow the link you posted above? Might there be a fix for that?

**Comment by dtroelofsprins** [comment]
> I noticed when not signed in I get endless spinner when I follow the link you posted above? Might there be a fix for that?

In EMX1 you get "Entity not found" if you don't have permissions to see the table, also not really informative, but at least you don't get a spinner. Ideally you would get something like 'permissions denied'.

**Review by dtroelofsprins — APPROVED** [review]
Can actually test on a preview server, but looking at the description of the email that will be sent it looks good!

File: backend/molgenis-emx2-tasks/src/main/java/org/molgenis/emx2/tasks/ScriptTask.java [hunk 1]
```diff
@@ -166,10 +168,15 @@ public void run() {
   private void sendFailureMail() {
     if (this.getFailureAddress() != null && !this.getFailureAddress().isEmpty()) {
       EmailService emailService = new EmailService();
```
[the remark this answers is not in the document]
Comment (harmbrugge) [reply backend/molgenis-emx2-tasks/src/main/java/org/molgenis/emx2/tasks/ScriptTask.java:171]: It's pretty lightweight and only init when a mail need to be send. Where would you put it?

File: backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/TaskApi.java [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import static org.molgenis.emx2.Constants.SYSTEM_SCHEMA;
 import static org.molgenis.emx2.FilterBean.f;
 import static org.molgenis.emx2.SelectColumn.s;
+import static org.molgenis.emx2.rdf.RDFUtils.extractHost;
```
[the remark this answers is not in the document]
Comment (harmbrugge) [reply backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/TaskApi.java]: Moved it to emx2

**Pull request merged**